nexus/tests/integration_tests/metrics.rs

```diff
@@ -399,6 +400,8 @@ async fn test_instance_watcher_metrics(
         )
         .await
         .unwrap();
+        // Make sure that the latest metrics have been collected.
+        oximeter.force_collect().await;
     };
```

Comment:
I'm looking at this change since it didn't seem to help with the flakiness -- I don't know that this would do anything? The line immediately following each use of the `activate_instance_watcher` closure is an awaited call to `timeseries_query`, which performs a `force_collect` first thing:

https://github.com/oxidecomputer/omicron/blob/e0a118496b05a30bf7a5a8a0820e0dda415c485a/nexus/tests/integration_tests/metrics.rs#L283-L288

Reply:
Huh, okay, I'm not sure if I understand why the test was flaky in the first place, then! I'll have to keep digging.

Reply:
Oh, yeah I definitely missed that. I've honestly found it frustrating to write metrics tests that assert exact equalities like this. There are just too many programs that the data has to move through, all of which are running a mess of Tokio tasks and the like (the producer, `oximeter`, ClickHouse, the test program, etc.). I've generally used inequalities if one wants to check things like the amount of data; or checks that there are samples with timestamps after a certain point; or some value exists in the data stream; etc.